sycl/source/detail/sycl_mem_obj_t.cpp

```diff
@@ -147,6 +147,17 @@ void SYCLMemObjT::determineHostPtr(const ContextImplPtr &Context,
   } else
     HostPtrReadOnly = false;
 }
+
+void SYCLMemObjT::detachMemoryObject(const std::shared_ptr<SYCLMemObjT> &self,
+                                     bool DefaultAllocator) const {
+  // Check MRecord without read lock because not found any usages that may bring
+  // corruption. MRecord is nullptr on buffer creation and set to meaningfull
```

Comment:
```suggestion
  // Check MRecord without read lock because at this point we expect that no commands that operate on the buffer can be created.
  // MRecord is nullptr on buffer creation and set to meaningfull
```

Reply:
fixed in https://github.com/intel/llvm/pull/6837/commits/28f008d324caa8739383fcf7915407c920f1ea98